x-pack/plugins/ml/public/application/model_management/get_model_state.tsx

```diff
@@ -5,13 +5,28 @@
  * 2.0.
  */
 
-import type { ModelState } from '@kbn/ml-trained-models-utils';
-import { MODEL_STATE } from '@kbn/ml-trained-models-utils';
+import { DEPLOYMENT_STATE, MODEL_STATE, type ModelState } from '@kbn/ml-trained-models-utils';
 import type { EuiHealthProps } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import type { ModelItem } from './models_list';
+
+/**
+ * Resolves result model state based on the download status and deployments stats.
+ */
+export const getModelDeploymentState = (model: ModelItem): ModelState | undefined => {
+  if (model.stats?.deployment_stats?.some((v) => v.state === DEPLOYMENT_STATE.STARTED)) {
+    return MODEL_STATE.STARTED;
+  }
+  if (model.stats?.deployment_stats?.some((v) => v.state === DEPLOYMENT_STATE.STARTING)) {
+    return MODEL_STATE.STARTING;
+  }
+  if (model.stats?.deployment_stats?.every((v) => v.state === DEPLOYMENT_STATE.STOPPING)) {
```

Comment:
it's not obvious that this is an `every` compared to the previous two `some` loops. Could you add a comment to point this out and to mention that every deployment must be in a stopping state for us to display the model as stopping.

Reply:
Added comments in https://github.com/elastic/kibana/pull/188847/commits/19af431b4af7b13fc064df6e3a9955ad16afc6e9

Reply:
Other models go into a `stopping` state for me when I stop an ELSER deployment:

<img width="1763" alt="Screenshot 2024-07-23 at 09 30 11" src="https://github.com/user-attachments/assets/8c47d6ba-f169-4c35-a6c4-946129c03a51">

Reply:
fixed in https://github.com/elastic/kibana/pull/188847/commits/97e063886ca99bff1cfdf9c7b86f25fd540c5089